src/java/org/apache/cassandra/cql3/Relation.java

```diff
@@ -203,7 +270,7 @@ protected abstract Restriction newSliceRestriction(TableMetadata table,
      * @return a new Contains <code>Restriction</code> instance
      * @throws InvalidRequestException if the <code>Relation</code> is not valid
      */
-    protected abstract Restriction newContainsRestriction(TableMetadata table, VariableSpecifications boundNames, boolean isKey);
+    protected abstract Restriction newContainsRestriction(TableMetadata table, VariableSpecifications boundNames, boolean isKey, boolean isNot);
```

Comment:
The new `isNot` parameter should probably be documented. However, given that EQ/NEQ and IN/NOT_INT pairs have separate `new*Restriction` methods for the "not" variant, perhaps we should do the same here and have a separate `newNotContains` method instead of the parameter?

Reply:
I added `newNotContainsRestriction` and removed the `isNot` param.